staging/src/k8s.io/apiextensions-apiserver/pkg/apis/apiextensions/types.go

```diff
@@ -177,6 +177,15 @@ type CustomResourceDefinitionVersion struct {
 	// Storage flags the version as storage version. There must be exactly one flagged
 	// as storage version.
 	Storage bool
+	// deprecated indicates this version of the custom resource API is deprecated.
+	// When set to true, API requests to this version receive a warning header in the server response.
+	// Defaults to false.
+	Deprecated bool
```

Comment:
Why don't we simply require someone to write their own DeprecationWarning and avoid having a bool?  I haven't read all the way down, but I doubt we have a very good stock message on these since versions don't follow kube versions.

Reply:
The normal case of "X is deprecated" or "X is deprecated; use Y instead" is easy to construct, and seemed valuable